lib/_http_outgoing.js

```diff
@@ -109,35 +109,29 @@ OutgoingMessage.prototype.setTimeout = function(msecs, callback) {
 // any messages, before ever calling this.  In that case, just skip
 // it, since something else is destroying this connection anyway.
 OutgoingMessage.prototype.destroy = function(error) {
-  if (this.socket)
+  if (this.socket) {
+    // If the socket is corked from a write, uncork it before destroying it.
+    if (this.socket._writableState.corked)
+      this.socket.uncork();
```

Comment:
which test was failing?

IMHO not transmitting the data when destroy() is hit is the correct behavior of destroy(). `destroy` is a dangerous operation anyway. @mafintosh what do you think?

Beware that, if the socket was corked twice, we need to call enough uncork() for this to take into effect.